server/sonar-qa-util/src/main/java/org/sonarqube/qa/util/pageobjects/RuleDetails.java

```diff
@@ -19,15 +19,251 @@
+  }
+
+  public RuleDetails deleteOnlyCustomRule() {
+    $$(".js-delete-custom-rule").shouldHaveSize(1).first().click();
+    modal().shouldBe(visible);
+    modal().find("button").click();
+    modal().shouldNotBe(visible);
+    return this;
+  }
+
+  public RuleDetails openOnlyCustomRule() {
+    $$(".coding-rules-detail-list-name a").shouldHaveSize(1).first().click();
+    return this;
+  }
+
+  public RuleActivation activateOnOnlyProfile() {
+    $("#coding-rules-quality-profile-activate").click();
+    modal().shouldBe(visible);
+    return new RuleActivation();
+  }
+
+  private SelenideElement modal() {
+    return $(".modal");
+  }
+
+  private SelenideElement takeCustomRuleByName(String ruleName) {
+    return $$(".coding-rules-detail-list-name").findBy(text(ruleName));
+  }
+
+  public ExtendedDescription extendDescription() {
+    return new ExtendedDescription().start();
+  }
+
+  public Tags tags() {
+    return new Tags();
+  }
+
+  public EditForm edit() {
+    $("#coding-rules-detail-custom-rule-change").click();
+    modal().shouldBe(visible);
+    return new EditForm();
+  }
+
+  public RuleActivation changeOnlyActivation() {
+    $$(".coding-rules-detail-quality-profile-change").shouldHaveSize(1).first().click();
+    modal().shouldBe(visible);
+    return new RuleActivation();
+  }
+
+  public RuleDetails onlyActivationShouldHaveParameter(String parameter, String value) {
+    $$(".coding-rules-detail-quality-profile-parameter")
+      .findBy(Condition.and("", text(parameter), text(value)))
+      .shouldBe(visible);
     return this;
   }
 
+  public RuleDetails onlyActivationShouldHaveSeverity(String severity) {
+    $(".coding-rules-detail-quality-profile-severity .icon-severity-" + severity.toLowerCase(Locale.ENGLISH)).shouldBe(visible);
+    return this;
+  }
+
+  public RuleDetails revertOnlyActivationToParentDefinition() {
+    $(".coding-rules-detail-quality-profile-revert").click();
+    $("button[data-confirm=\"yes\"").click();
+    return this;
+  }
+
+  public static class ExtendedDescription {
+    public ExtendedDescription start() {
+      $("#coding-rules-detail-extend-description").click();
+      getTextArea().shouldBe(visible);
+      return this;
+    }
+
+    public ExtendedDescription cancel() {
+      $("#coding-rules-detail-extend-description-cancel").click();
+      getTextArea().shouldNotBe(visible);
+      return this;
+    }
+
+    public ExtendedDescription type(String text) {
+      getTextArea().val(text);
+      return this;
+    }
+
+    public ExtendedDescription submit() {
+      $("#coding-rules-detail-extend-description-submit").click();
+      getTextArea().shouldNotBe(visible);
+      return this;
+    }
+
+    public ExtendedDescription remove() {
+      $("#coding-rules-detail-extend-description-remove").click();
+      $("button[data-confirm=\"yes\"").click();
+      getTextArea().shouldNotBe(visible);
+      return this;
+    }
+
+    private SelenideElement getTextArea() {
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Make "getTextArea" a "static" method. ([squid:S2325](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS2325))

[See it in SonarQube](https://next.sonarqube.com/sonarqube/project/issues?id=org.sonarsource.sonarqube%3Asonarqube&branch=feature%2Fstas%2Fit-rules&issues=AWEyHl7czwlWAzALxAmY&open=AWEyHl7czwlWAzALxAmY)

[ISSUE_KEY]: AWEyHl7czwlWAzALxAmY